client/app/components/game-client.js

```diff
@@ -0,0 +1,64 @@
+import Ember from 'ember';
+import GameChannel from 'vilive/utils/game-channel';
+
+export default Ember.Component.extend({
+  init() {
+    this.set('messages', Ember.makeArray());
+    GameChannel.init(this.get('session-manager').currentToken());
+    GameChannel.on('event', payload => {
+      this.putMessage(true, payload.body);
+      this.scrollToBottomMessages();
+    });
+    this._super(...arguments);
+  },
+
+  actions: {
+    sendCommand() {
+      const command = this.get('command');
+
+      if (!command || command.length === 0) { return; }
+
+      // hack for template at the moment
+      // false means cliean, true means server
+      this.putMessage(false, command);
+      this.scrollToBottomMessages();
+
+      GameChannel.push('user_cmd', {body: command});
+      this.set('command', null);
+    }
+  },
+
+  /**
+    This does not fire after each message is inserted just when the component
+    is inserted, in here we are going to focus on the text box.
+  */
+  didInsertElement() {
+    this.$('#command-input').focus();
+  },
+
+  /**
+    Need to scroll the div after the message has been rendered.
+    otherwise it stops short and scroll to last message inserted
+  */
+  scrollToBottomMessages() {
+    const div = this.$('.panel-body');
+    Ember.run.schedule('afterRender', () => {
```

Comment:
From what i have read this is right - I wanted to do it automatically on `didInsertElement` but that seems only fire when the component is inserted, not when elements within a compent are changed from my testing. Docs are sparse on the `didInsertElement` and not even listed directly on component.

None the less this solved the issue I had with scrolling where it was one message behind essentially as the current message had not been rendered.